src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/MessageboxInstancesController.cs

```diff
@@ -215,14 +215,15 @@ public async Task<ActionResult> Undelete(int instanceOwnerPartyId, Guid instance
                 return StatusCode(500, $"Unknown database exception in restore: {dce}");
             }
 
-            if (instance.Status.HardDeleted.HasValue)
+            if (instance.Status.IsHardDeleted)
             {
                 return BadRequest("Instance was permanently deleted and cannot be restored.");
```

Comment:
Should we change from a BadRequest to NotFound response here?

The current message kind of indicate that the instance isn't completely gone yet. Good or bad? Maybe use a more standard not found message.

Reply:
Updated to notFound. I don't think the message is a big issue. Endpoint is intended for sbl, so no one will actually read it, but if they do I guess they would know there is a way to manually restore if they actually need to retrieve the data